### ISSUE TITLE
Torch to TOSA conversion fails to legalize 'torch.constant.int'

### ISSUE THREAD

**Issue by Svoch** [llvm/torch-mlir#961]
I am trying to compile a portion of a PyTorch Self-Attention module down to TOSA backend and am hitting an error on legalizing the `torch.contant.int` Op in TOSA conversion pass. The issue raises only when output type is set to `torch_mlir.OutputType.TOSA` in Torch-MLIR compile API. The conversion to LinAlg Dialect and further down to backend works fine. However Torch to TOSA conversion is failing.

### Error log
```
Exception: 
Lowering Torch Backend IR -> TOSA Backend IR failed with the following diagnostics:
error: failed to legalize operation 'torch.constant.int'
note: see current operation: %5 = "torch.constant.int"() {value = 0 : i64} : () -> !torch.int
```
### Steps to reproduce
The [script](https://github.com/Svoch/torch-mlir/blob/deb5d7ae59c63b93d9ffb865a77c1455bf0d51bb/examples/attention_scores.py) to reproduce the error is up in [this draft PR](https://github.com/Svoch/torch-mlir/pull/1) on a local fork. The error can be reproduced using the code snippet below with the module definition and `torch_mlir.compile()` API call:
```
class AttentionScores(torch.nn.Module):
    def __init__(self, embedding_dim, num_heads):
        super(AttentionScores, self).__init__()
        self.query = nn.Linear(embedding_dim, num_heads)
        self.key = nn.Linear(embedding_dim, num_heads)

    def forward(self, inputs):
        query = self.query(inputs)
        key = self.key(inputs)
        scores = torch.matmul(query, key.transpose(0, 1))
        return scores

attention_scores = AttentionScores(embedding_dim=10, num_heads=2)
inputs = torch.rand(5, 10)
tosa_module = torch_mlir.compile(attention_scores, inputs, output_type=torch_mlir.OutputType.TOSA)
```

This issue is potentially relevant to what @nithinsubbiah, @rdadolf and I are seeing in #910. I was also able to reproduce the error by simplifying the module above to a single Transpose Op (i.e. `torch.Tensor.transpose` in forward-propagate method).

cc @sjarus @powderluv @silvasean - wonder if you have seen this or have any insight on what might have been going wrong here.

**Comment by sjarus** [comment]
I've encountered this already @Svoch - it also impacts MobilenetsV3 . Working on a fix internally but am getting some BERT ones out first.

**Comment by YellowHCH** [comment]
I just convert ConstantIntOp to arith while converting Bert to tosa， as a intermediate result.

**Comment by Svoch** [comment]
Below is the IR I get lowering the model above to Torch Dialect. If I've understood correctly, even though rewriters for [AtenMmOp](https://github.com/llvm/torch-mlir/blob/5888c4f7dc5aad504ef8c77442ebbf3c0393623c/lib/Conversion/TorchToTosa/TorchToTosa.cpp#L1497) and [AtenLinearOp](https://github.com/llvm/torch-mlir/blob/5888c4f7dc5aad504ef8c77442ebbf3c0393623c/lib/Conversion/TorchToTosa/TorchToTosa.cpp#L1533) do exist in TorchToTosa lowering pass, there is no lowering pattern for `AtenTransposeIntOp`. Seems like this is the underlying problem here. Does it make sense @sjarus?
```
module attributes {torch.debug_module_name = "AttentionScores"} {
  func.func @forward(%arg0: !torch.vtensor<[5,10],f32>) -> !torch.vtensor<[5,5],f32> {
    %0 = torch.vtensor.literal(dense<[0.143874153, 0.230392322]> : tensor<2xf32>) : !torch.vtensor<[2],f32>
    %1 = torch.vtensor.literal(dense<[[0.226313293, -0.279624343, -0.211782753, 2.701980e-01, 0.0410803184, 0.25729695, -0.00262214779, -0.0828355625, 0.145104617, -0.0266586915], [0.239182726, 0.00810842216, -0.00369983795, -0.132520735, -0.254919976, 0.0812480971, -0.196122929, 0.10878253, 0.158111736, 0.306294829]]> : tensor<2x10xf32>) : !torch.vtensor<[2,10],f32>
    %2 = torch.vtensor.literal(dense<[0.249187589, 0.0799354762]> : tensor<2xf32>) : !torch.vtensor<[2],f32>
    %3 = torch.vtensor.literal(dense<[[2.573700e-01, 0.215394989, 0.238999024, 0.163943127, 0.212515891, 0.231857046, -0.28136012, -0.118400358, -0.217035949, -0.0496219955], [-0.3136262, 0.105287395, -0.037995059, -0.129876241, -0.142800108, -0.208000481, -0.0777831152, 0.144313246, -0.086798042, 0.255681425]]> : tensor<2x10xf32>) : !torch.vtensor<[2,10],f32>
    %int0 = torch.constant.int 0
    %int1 = torch.constant.int 1
    %4 = torch.aten.linear %arg0, %3, %2 : !torch.vtensor<[5,10],f32>, !torch.vtensor<[2,10],f32>, !torch.vtensor<[2],f32> -> !torch.vtensor<[5,2],f32>
    %5 = torch.aten.linear %arg0, %1, %0 : !torch.vtensor<[5,10],f32>, !torch.vtensor<[2,10],f32>, !torch.vtensor<[2],f32> -> !torch.vtensor<[5,2],f32>
    %6 = torch.aten.transpose.int %5, %int0, %int1 : !torch.vtensor<[5,2],f32>, !torch.int, !torch.int -> !torch.vtensor<[2,5],f32>
    %7 = torch.aten.mm %4, %6 : !torch.vtensor<[5,2],f32>, !torch.vtensor<[2,5],f32> -> !torch.vtensor<[5,5],f32>
    return %7 : !torch.vtensor<[5,5],f32>
  }
}
```

**Comment by Shukla-Gaurav** [comment]
I am encountering same issue with this patch https://github.com/llvm/torch-mlir/pull/862 for ResNet18 static model. @sjarus

**Comment by AmosLewis** [comment]
I encounter the same issue to lower the huggingface gpt2. [https://gist.github.com/AmosLewis/9b929414d5677afda3528122f92bbc73](https://gist.github.com/AmosLewis/9b929414d5677afda3528122f92bbc73)  @sjarus 
error: failed to legalize operation 'torch.constant.int'

**Comment by sjarus** [comment]
torch.constant.int is a known missing conversion.

**Comment by Svoch** [comment]
@silvasean - Yes. with the Torch to TOSA conversion of the Transpose Op merged, this issue can be marked as fixed.

Please note that the symptom, i.e. `error: failed to legalize operation 'torch.constant.int'` is common for most of the Ops without a TOSA lowering in Torch-MLIR, since they usually will leave dangling attributes (such as axis integers) after the Torch to TOSA conversion.
